zulipterminal/ui_tools/views.py

```diff
@@ -288,6 +288,25 @@ def update_streams(self, search_box: Any, new_text: str) -> None:
             ]
             streams_display = match_stream(stream_buttons, new_text,
                                            self.view.pinned_streams)[0]
+
+            # Add a divider to separate pinned streams from the rest.
+            pinned_stream_names = [
+                stream[0]
+                for stream in self.view.pinned_streams
+            ]
+            first_unpinned_index = len(streams_display)
+            for index, stream in enumerate(streams_display):
+                if stream.stream_name not in pinned_stream_names:
+                    first_unpinned_index = index
+                    break
+            if (first_unpinned_index != 0
+                    and first_unpinned_index != len(streams_display)):
+                # Check whether a divider already exists.
+                if not isinstance(streams_display[first_unpinned_index],
+                                  urwid.Divider):
```

Comment:
`StreamsViewDivider`?

Also, I remember you discussed the need for this section, but could you explain how it can happen in the comment, ie. what situation you're avoiding?